src/components/SuperDrawer/useSuperDrawer.tsx

```diff
@@ -64,40 +64,48 @@ export function useSuperDrawer(): UseSuperDrawerHook {
   } = useContext(BeamContext);
   const { openModal } = useModal();
 
+  /**
+   * Validates whether a new Superdrawer can be opened / whether the current Superdrawer can be closed.
+   * Breaking the validation into a separate function allows openInDrawer
+   * to validate the drawer's state without closing it
+   * */
+  function validateDrawerChange (onClose: () => void) {
+    const contentStackLength = contentStack.current.length;
+    if (contentStackLength === 0) {
+      return true;
+    }
+
+    // Attempt to close each drawer details
+    for (let i = contentStackLength - 2; i >= 0; i--) {
+      for (const canCloseDrawerDetail of canCloseDetailsChecks.current[i] ?? []) {
+        if (!canCloseDrawerDetail()) {
+          openModal({
+            title: "Confirm",
+            content: <ConfirmCloseModal onClose={onClose} />,
+          });
+          return false;
```

Comment:
Might be able to move this into it's own function to also be used in the `closeDrawerDetail`

Reply:
@KoltonG I realized we get into the same issue of needing to return a `Boolean` from a broken out function, since if we broken this out there would be two function responsible for calling doChange. The child function, called within `maybeChangeDrawer` would need to tell `maybeChangeDrawer` to break as well, otherwise `doChange` could get called twice. 

I might leave out this refactor for the time being at least.

Reply:
Eh, scratch that. Even with the Boolean pattern, this feels clearer
```
    for (let i = contentStackLength - 2; i >= 0; i--) {
      if (!canCloseDrawerDetails(i, doChange)) {
        return;
      }
    }
```